integration-tests/src/test/java/org/apache/fineract/integrationtests/ExternalBusinessEventTest.java

```diff
@@ -297,6 +342,31 @@ && loanTermVariationsMatch((List<Map<String, Object>>) externalEvent.getPayLoad(
         }
     }
 
+    public void verifyLoanTransactionBusinessEvents(BusinessEvent... businessEvents) {
+        List<ExternalEventDTO> allExternalEvents = ExternalEventHelper.getAllExternalEvents(requestSpec, responseSpec);
+        logBusinessEvents(allExternalEvents);
+        Assertions.assertNotNull(businessEvents);
+        Assertions.assertNotNull(allExternalEvents);
+        Assertions.assertTrue(businessEvents.length <= allExternalEvents.size(),
+                "Expected business event count is less than actual. Expected: " + businessEvents.length + " Actual: "
+                        + allExternalEvents.size());
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATETIME_PATTERN, Locale.ENGLISH);
+        for (BusinessEvent businessEvent : businessEvents) {
+            long count = allExternalEvents.stream().filter(externalEvent -> {
+
+                Object netDisbursalAmount = externalEvent.getPayLoad().get("netDisbursalAmount");
+                Object outstandingLoanBalance = externalEvent.getPayLoad().get("outstandingLoanBalance");
+                var businessDate = LocalDate.parse(businessEvent.getBusinessDate(), formatter);
+
+                return Objects.equals(externalEvent.getType(), businessEvent.getType())
+                        && Objects.equals(externalEvent.getBusinessDate(), businessDate)
+                        && Objects.equals(netDisbursalAmount, businessEvent.getPrincipalDisbursed())
```

Comment:
it does not check neither transaction amount, neither portions... Would you mind to change it to fit better for validating transaction business event?

Reply:
make sense, ok